pkg/types/stackerfile.go

```diff
@@ -58,28 +58,69 @@ func (sf *Stackerfile) Len() int {
 }
 
 func substitute(content string, substitutions []string) (string, error) {
+	// replace all placeholders where we have a substitution provided
+	sub_usage := []int{}
 	for _, subst := range substitutions {
 		membs := strings.SplitN(subst, "=", 2)
 		if len(membs) != 2 {
 			return "", errors.Errorf("invalid substition %s", subst)
 		}
-
-		from := fmt.Sprintf("$%s", membs[0])
 		to := membs[1]
 
-		log.Debugf("substituting %s to %s", from, to)
+		// warn on finding unsupported placeholders matching provided
+		// substitution keys:
+		nobracket := fmt.Sprintf("$%s", membs[0])
+		onebracket := fmt.Sprintf("${%s}", membs[0])
+		bad_content := content
+		for _, unsupp_placeholder := range []string{nobracket, onebracket} {
+			bad_content := strings.Replace(content, unsupp_placeholder, to, -1)
```

Comment:
Wouldn't strings.Index(unsupp_placeholder) != -1 work as well and be faster?

Reply:
yep

Reply:
I changed this in the second commit just now